pkg/storage/ingest/reader.go

```diff
@@ -346,24 +348,38 @@ func (r *partitionCommitter) run(ctx context.Context) error {
 				continue
 			}
 			previousOffset = currOffset
-
-			toCommit := kadm.Offsets{}
-			// Commit the offset after the last record.
-			// The reason for this is that we resume consumption at this offset.
-			// Leader epoch is -1 because we don't know it. This lets Kafka figure it out.
-			toCommit.AddOffset(r.kafkaCfg.Topic, r.partitionID, currOffset+1, -1)
-
-			committed, err := r.admClient.CommitOffsets(ctx, consumerGroup, toCommit)
-			if err != nil || !committed.Ok() {
-				level.Error(r.logger).Log("msg", "encountered error while committing offsets", "err", err, "commit_err", committed.Error(), "offset", currOffset)
-			} else {
-				committedOffset, _ := committed.Lookup(r.kafkaCfg.Topic, r.partitionID)
-				level.Debug(r.logger).Log("msg", "committed offset", "offset", committedOffset.Offset.At)
-			}
+			r.commit(ctx, currOffset)
 		}
 	}
 }
 
+func (r *partitionCommitter) commit(ctx context.Context, offset int64) {
+	toCommit := kadm.Offsets{}
+	// Commit the offset after the last record.
+	// The reason for this is that we resume consumption at this offset.
+	// Leader epoch is -1 because we don't know it. This lets Kafka figure it out.
+	toCommit.AddOffset(r.kafkaCfg.Topic, r.partitionID, offset+1, -1)
+
+	committed, err := r.admClient.CommitOffsets(ctx, consumerGroup, toCommit)
+	if err != nil || !committed.Ok() {
+		level.Error(r.logger).Log("msg", "encountered error while committing offsets", "err", err, "commit_err", committed.Error(), "offset", offset)
+	} else {
+		committedOffset, _ := committed.Lookup(r.kafkaCfg.Topic, r.partitionID)
+		level.Debug(r.logger).Log("msg", "committed offset", "offset", committedOffset.Offset.At)
+	}
+}
+
+func (r *partitionCommitter) stop(error) error {
+	offset := r.toCommit.Load()
+	if offset == noOffset {
+		return nil
+	}
+	ctx, cancel := context.WithTimeoutCause(context.Background(), 10*time.Second, fmt.Errorf("timed out while doing final partition commit"))
+	defer cancel()
```

Comment:
Why do we need this? Aren't we covered by the default timeouts involved in `CommitOffset()`? Putting something hardcoded here looks more hacky to me.

Reply:
it felt easier to reason that the call to CommitOffset won't take ages to complete and won't block shutdown. Removed the timeout and added a comment instead